lib/google/serverless/exec.rb

```diff
@@ -415,15 +418,18 @@ def new_rake_task name, args: [], env_args: [],
       #     image to use. (Applies only to the "cloud_build" strategy.)
       # @param strategy [String,nil] The execution strategy to use, or `nil` to
       #     choose a default based on the App Engine environment (flexible or standard) or
-      #     Cloud Run environemnts. Allowed values are `nil`, `"deployment"` (which is the
+      #     Cloud Run environments. Allowed values are `nil`, `"deployment"` (which is the
       #     default for App Engine Standard), and `"cloud_build"` (which is the default for
       #     App Engine Flexible and Cloud Run).
       # @param gcs_log_dir [String,nil] GCS bucket name of the cloud build log
       #     when strategy is "cloud_build". (ex. "gs://BUCKET-NAME/FOLDER-NAME")
+      # @param product [Symbol] The serverless product. If omitted, defaults to the
+      #     value returns by {Google::Serverless::Exec.default_product}.
+      #     Allowed values are `APP_ENGINE` and `CLOUD_RUN`.
       #
       def initialize command,
                      project: nil, service: nil, config_path: nil, version: nil,
-                     timeout: nil, wrapper_image: nil, strategy: nil, gcs_log_dir: nil
+                     timeout: nil, wrapper_image: nil, strategy: nil, gcs_log_dir: nil, product: nil
```

Comment:
@dazuma when you mentioned the `product` can't be `nil`, does that mean I have to change the initializer also? Right now, `default_product` is being called in `resolve_parameters`, but wondering if I'd have to switch it?

Reply:
Oh, looks like I was wrong on this. The usage is that `nil` actually is allowed, and then it gets resolved one way or the other when `resolve_parameters` is called. My bad.